mobile/src/main/java/org/openhab/habdroid/background/ItemsControlsProviderService.kt

```diff
@@ -140,17 +140,22 @@ class ItemsControlsProviderService : ControlsProviderService() {
                 return@launchWithConnection
             }
 
-            itemNames.forEach { itemName ->
-                try {
-                    ItemClient.loadItem(connection, itemName)?.let { item ->
-                        maybeCreateControl(item, primaryServerName, true)?.let { control ->
-                            items[item.name] = item
-                            publisher.add(control)
-                        }
-                    }
-                } catch (e: HttpClient.HttpException) {
-                    Log.e(TAG, "Could not load item $itemName", e)
+            val allItems = try {
+                ItemClient.loadItems(connection) ?: emptyList()
+            } catch (e: HttpClient.HttpException) {
+                Log.e(TAG, "Could not load items", e)
+                emptyList()
+            }
+
+            itemNames
+                .map { itemName ->
+                    allItems.first { item -> item.name == itemName }
                 }
+                .forEach { item ->
+                    maybeCreateControl(item, allItems, primaryServerName, true)?.let { control ->
+                        items[item.name] = item
```

Comment:
Seeing this now: could probably move this out and just create the map from `allItems`.

Reply:
What exactly do you mean with that?

Reply:
Pull filling `items` out of the `forEach` and fill it separately, using a simple `map`.

Reply:
Can even remove `items` completely and make `allItems` a `Map<String, Item>`.

Reply:
Basically this: https://github.com/maniac103/openhab.android/commit/9da126d1502242e6f94e17179a26108c9c81f76f

Reply:
Done: https://github.com/openhab/openhab-android/pull/2925/commits/3e16b0a8c59b4c65dc1c8f76b4e34e43fed13474